reactor-core/src/main/java/reactor/core/scheduler/SingleScheduler.java

```diff
@@ -54,8 +54,11 @@ final class SingleScheduler implements Scheduler, Supplier<ScheduledExecutorServ
 					SingleScheduler.class, SchedulerState.class, "state"
 			);
 
+	private static final SchedulerState<ScheduledExecutorService> INIT =
+			SchedulerState.init(TERMINATED);
+
 	SingleScheduler(ThreadFactory factory) {
-		this.state = SchedulerState.init(TERMINATED);
+		this.state = INIT;
```

Comment:
this can be replaced with `STATE.lazySet` as a cheper operation, also this statement should be very last in the line